src/core/server/http/http_config.ts

```diff
@@ -38,7 +39,17 @@ export const config = {
           validate: match(validBasePathRegex, "must start with a slash, don't end with one"),
         })
       ),
-      defaultRoute: schema.maybe(schema.string()),
+      csp: schema.object(
+        {
+          directives: schema.string(),
+          rules: schema.arrayOf(schema.string()),
+          strict: schema.boolean(),
+          warnLegacyBrowsers: schema.boolean(),
+        },
+        {
+          defaultValue: DEFAULT_CSP_OPTIONS,
+        }
```

Comment:
I used this style as it was demonstrated in the HTTP service:

https://github.com/elastic/kibana/blob/21edc28ed3d49f4347761c94f284d534a3061c07/src/core/server/http/http_config.ts#L50-L52

Reply:
- when I try to configure any value in `kibana.yml` I received the exception: 
```
 FATAL  ValidationError: child "server" fails because ["csp" is not allowed]
```
- are you sure that this syntax allows users to specify **only one** rule and defaults applied for the remaining settings?